libpromises/matching.c

```diff
@@ -308,6 +308,27 @@ void EscapeSpecialChars(const char *str, char *strEsc, int strEscSz, char *noEsc
     }
 }
 
+size_t EscapeRegexCharsLen(const char *str)
+{
+    size_t ret = 0;
+    for (const char *sp = str; *sp != '\0'; sp++)
+    {
+        switch (*sp)
+        {
+            case ' ':
```

Comment:
Shouldn't this be ``'.'``, like in the function below?

Both functions are actually quite bad at escaping regex patterns, there are lots of other special characters too: ``[, ], ?, (, )``, not to mention ``\`` itself. But at least this way they are in sync.